contracts/lib/PepemonConfig.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./AdminRole.sol";
+import "../iface/IConfigurable.sol";
+
+/**
+ * @notice This contract stores the addresses of all and any other contracts used by Pepemon
+ * @dev This contract must be added as an Admin on contracts before "syncContractConfig" can be called.
+ */
+contract PepemonConfig is AdminRole {
+    struct ContractDisplayData {
+        address contractAddress;
+        string contractName;
+    } 
+
+    string[] private contactsNames;
+    mapping(string => address) public contractAddresses;
+
+    /**
+     * @notice Adds or updates contracts addresses associated by contract names
+     * @param contractName Name of the contract that will be stored
+     * @param contractAddress Address of the contract that will be stored
+     * @param callSync When true, the function "syncConfig" of the contract being stored will be invoked
+     */
+    function setContractAddress(string calldata contractName, address contractAddress, bool callSync) external onlyAdmin {
+        require(contractAddress != address(0));
+
+        // If its the first time adding the contract, store its name in the array
+        if (contractAddresses[contractName] == address(0)) {
+            contactsNames.push(contractName);
+        }
+        contractAddresses[contractName] = contractAddress;
+        if (callSync) {
+            IConfigurable(contractAddress).syncConfig();
+        }
+    }
+
```

Comment:
Setting multiple contracts in a single tx would be nice, could prevent some dependency issues